vexcl/backend/common.hpp

```diff
@@ -244,16 +244,42 @@ class sha1_hasher {
         }
 
         operator std::string() {
-            unsigned int digest[5];
+            boost::uuids::detail::sha1::digest_type digest;
             h.get_digest(digest);
-
+            return serialize_digest(digest);
+        }
+    private:
+        // Before boost 1.86.0, boost::uuids::detail::sha1 used to be unsigned int[5].
+        std::string serialize_digest(unsigned int digest[5]) const {
             std::ostringstream buf;
             for(int i = 0; i < 5; ++i)
                 buf << std::hex << std::setfill('0') << std::setw(8) << digest[i];
+            return buf.str();
+        }
+
+        // Since boost 1.86.0, boost::uuids::detail::sha1 has been unsigned char[20].
+        std::string serialize_digest(unsigned char digest[20]) const {
+            // Convert digest to a string representation of 5 integers in hexadecimal format
+            // (digest_type used to be unsigned int[5]). The reason for boost's change is to
+            // fix a bug related to endianness, but to conform to old VexCL behavior, bytes
+            // are reordered if necessary (whether conformance is required is up for debate).
```

Comment:
I don't think this is important. VexCL cache is stored and used on the same machine it is generated on, so the only important thing is the lack of collisions. The worst that could happen when someone upgrades to this version, is cache duplication (in case the old and the new hashes have opposing byte order).